### ISSUE TITLE
Feature Request: AV8B Harrier Targetpoints instead of waypoints

### ISSUE THREAD

**Issue by danw-eth** [aronCiucu/DCSTheWay#137]
In the Harrier we can't assign JDAMs to waypoints directly, they can only use target points.
The process to convert waypoints to target points is rather tedious due to the Harriers new logic for switching between nav point types.

By default most people would prefer target points than waypoints (in my opinion) when it comes to the Harrier, and they can be created directly with coordinates just like the waypoints can. I think you need to have at least one target point first though, which can be created from the existing waypoint if necessary by designating it then using TOO. You then hold down the waypoint up or down button to switch between waypoints and target point mode.

Would it be possible to read the current state and enter points based on that? i.e. if the cockpit nav system has TP1 selected then enter the data as target points, if it has WP1 selected then enter as waypoints.
Alternatively a simple preference setting stating that we prefer either TP or WP would be great.

Thanks!

**Comment by snakepl88** [comment]
It would be nice to toggle between TP or WP. For now there is a way to import target points from the F10 map using the CAS page in the harrier. Here is a video: https://www.youtube.com/watch?v=Cg4gimQxT6Q

**Comment by danw-eth** [comment]
Yes, the F10 map method is what I use currently.

But I much prefer the consistent method we use across all aircraft in TheWay.
Especially if they are strike targets and not CAS targets.

**Comment by snakepl88** [comment]
> But I much prefer the consistent method we use across all aircraft in TheWay.

Agree, also I noticed that the program asks if you are in pilot or gunner seat in the AH64, so the same logic can be applied and it can ask if you want WP or TP before transferring.

Some things to note: I think you can have a total of 10 Target Points max, T1-T9 and TX, so the best way might be start with target point 1 and overwrite all the way to 10 and then stop if there are more than 10 points selected on map.

**Comment by snakepl88** [comment]
I think I have it working, try it out. Adds a dialog option to choose waypoints or targetpoints in the AV8B.
https://github.com/snakepl88/DCSTheWay/releases

But I don't think this is a good permanent solution. As this program gets more complex and more modules are added, a better solution might be to add a modifier type to a waypoint on the list (for example: id, name, _**type**_, lat, long, elev). The _**type**_ then could be WPT, TGPT, MKPT, etc. that then can be used across different modules, just a thought.

**Comment by danw-eth** [comment]
Awesome thanks!

Agree yes not ideal for permanent solution, because can't see the dialog in VR.

I think a separate keybind to "send targetpoints to jet" so that you can send either one without having to takeoff the headset. But agree it would be best to be consistent across different modules that also have targetpoints or markpoints.

Could have the type on the line as well, but then would need to add yet another keybind that does "save targetpoint". The KISS solution would be only to consider the type when you send it to the jet, and just use the delete keybind in between setting up waypoints and targetpoints.

**Comment by snakepl88** [comment]
I just found a major issue. Harrier only accepts manually entered coordinates in degrees, minute, seconds. So, the target points made using this method are not accurate enough for JDAM bombing. JDAMs will miss by a bit. Test it on your end but I'm pretty sure that's the issue.

The only accurate way of doing this is still probably DCS F10 map method.

**Comment by danw-eth** [comment]
You need to use decimal to add more precision. 
See about halfway through this video:
https://youtu.be/9zm_nLhFOqA?si=U2bON38Sd1urxZ9b

**Comment by snakepl88** [comment]
I'm going to take a look into this. Looks like this was added fairly recently. This might be an easy fix.

**Comment by danw-eth** [comment]
Thanks, yes that was a new feature in one of Razbam's patches a few months ago in open beta.

**Comment by snakepl88** [comment]
Changed av8b to accept more accurate coordinates.
Works correctly now, check the link I posted before for the release.

**Comment by danw-eth** [comment]
Awesome thanks!
Not doing much DCS at the moment because of work leading up to Christmas, but will set a reminder to update to this in the new year when I get back from travels thanks

**Comment by aronCiucu** [comment]
Good work. i will also think about a new solution UI-wise to allow for other types of waypoints

**Comment by jonsky752** [comment]
@snakepl88 Tried it and it works great, thanks
